src/main/sensors/battery.c

```diff
@@ -232,16 +233,48 @@ static void batteryUpdateVoltageState(void)
         default:
             break;
     }
+
+}
+
+static void batteryUpdateLVC(timeUs_t currentTimeUs) 
+{   
+    if (batteryConfig()->lvcEnabled) {
+        if (voltageState == BATTERY_CRITICAL && !lowVoltageCutoff.enabled) {
+            lowVoltageCutoff.enabled = true;
+            lowVoltageCutoff.startTime = currentTimeUs;
+            lowVoltageCutoff.percentage = 100;
+        }
+        if (lowVoltageCutoff.enabled) {
+            if (currentTimeUs - lowVoltageCutoff.startTime < LVC_AFFECT_TIME) {
+                lowVoltageCutoff.percentage = 100 - ((currentTimeUs - lowVoltageCutoff.startTime) * (100 - batteryConfig()->lvcPercentage) / LVC_AFFECT_TIME); 
```

Comment:
I'll switch to using cmp32.

Was there a slower clock I could grab from some where? Maybe we can switch to 64bit time for the microsec clock :)